scipy/stats/tests/test_stats.py

```diff
@@ -2469,6 +2471,14 @@ def test_onesample(self):
             assert_raises(ValueError, stats.ttest_1samp, x, 5.0,
                           nan_policy='foobar')
 
+    def test_1samp_alternative(self):
+        assert_raises(ValueError, stats.ttest_1samp, self.X1, 0, alternative="error")
+
+        t, p = stats.ttest_1samp(self.X1, 1, alternative="less")
+        assert_array_almost_equal(p, self.P1_1_l)
+
+        t, p = stats.ttest_1samp(self.X1, 1, alternative="greater")
+        assert_array_almost_equal(p, self.P1_1_g)
```

Comment:
Also, I haven't looked at the two-sided tests - does it have any for when the input data is a 2d array? I wouldn't say you have to fix it if not, but you do need to confirm that what you wrote works for 2d arrays.
See the [documentation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_1samp.html).
So you'd need to test something like:
```
ttest_1samp([[0, 1, 2], [-1, 0, 1], [-2, -1, 0]], 0, axis=1, alternative='greater')
```
and separately 
```
ttest_1samp([[0, 1, 2], [0, 1, 2]], 0, axis=None, alternative='greater')
```
because the latter tests over the whole array.

Weird that the documentation doesn't mention the default... I would suggest adding that information.

Reply:
Yep, it looks like it does (in test_ttest_1samp_new() at line 3717 - rvn1 is a multi-dimensional array). I've also tested my implementation and it seems to work fine.

Reply:
OK, please add to `test_ttest_1samp_new` a test of your `alternative` keyword as suggested above. Also, please add to the `ttest_1samp` documentation what the default axis is.

Reply:
For example:
```
axis : int or None, optional
        Axis along which to compute test; default is ___. If None, compute over the whole
        array `a`.
```
I haven't checked what the default is.

It's not your fault that it's not already documentated, but it would be nice to fix it while we're working on this.